modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/translation/TranslationModal.js

```diff
@@ -0,0 +1,264 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAlert from '@clayui/alert';
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayLink from '@clayui/link';
+import ClayManagementToolbar from '@clayui/management-toolbar';
+import ClayModal from '@clayui/modal';
+import ClayTable from '@clayui/table';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const TranslationModal = ({
+	ariaLabels = {
+		default: Liferay.Language.get('default'),
+		manageTranslations: Liferay.Language.get('manage-translations'),
+		managementToolbar: Liferay.Language.get('management-toolbar'),
+		translated: Liferay.Language.get('translated'),
+		untranslated: Liferay.Language.get('untranslated'),
+	},
+	defaultLocale,
+	locales,
+	observer,
+	translations,
+}) => {
+	const [active, setActive] = React.useState(false);
+	const [deletedLocale, setDeletedLocale] = React.useState(null);
+	const [deletedLocales, setDeletedLocales] = React.useState(new Set([]));
+	const [initialLocales, setInitialLocales] = React.useState(locales);
+	const [value, setValue] = React.useState('');
+
+	const filteredLocales = React.useMemo(() => {
+		return initialLocales.filter((locale) =>
+			locale.label.match(new RegExp(value, 'i'))
+		);
+	}, [initialLocales, value]);
+
+	const restoreLocale = (locale) => {
+		if (deletedLocales.has(locale)) {
+			deletedLocales.delete(locale);
+			setDeletedLocales(deletedLocales);
+		}
+
+		filteredLocales.splice(locale.index, 0, locale);
+		setInitialLocales(filteredLocales);
+		setActive(false);
+		setDeletedLocale(null);
```

Comment:
I think the same situation compared here above, we can avoid changing `filteredLocales` and also use it to update the `initialLocales`.

```js
setInitialLocales([...initialLocales, locale]);
setActive(false);
setDeletedLocale(null);
```